app/src/pipelines/generate_referrals/pipeline_wrapper.py

```diff
@@ -66,21 +85,33 @@ def run_api(self, query: str, prompt_version_id: str = "") -> dict:
             prompt_template = haystack_utils.get_phoenix_prompt(
                 "generate_referrals", prompt_version_id
             )
+        except httpx.HTTPStatusError as he:
+            raise HTTPException(
+                status_code=422,
+                detail=f"The requested prompt version '{prompt_version_id}' could not be retrieved",
+            ) from he
+
+        try:
+            output_validator = self.pipeline.get_component("output_validator")
+            assert isinstance(output_validator, components.LlmOutputValidator)
+            # Reset attempt counter
+            output_validator.attempt_count = 0
+
             response = self.pipeline.run(
                 {
                     "prompt_builder": {
                         "template": prompt_template,
                         "query": query,
-                        "resource_json": resource_as_json,
+                        "response_json": response_schema,
                     },
                 },
                 include_outputs_from={"llm", "save_result"},
             )
             logger.info("Results: %s", pformat(response, width=160))
             return response
+        except PipelineRuntimeError as re:
+            logger.error("PipelineRuntimeError: %s", re, exc_info=True)
+            raise HTTPException(status_code=500, detail=str(re)) from re
         except Exception as e:
-            logger.error("The requested prompt version could not be retrieved")
-            raise HTTPException(
-                status_code=400,
-                detail=f"The requested prompt version '{prompt_version_id}' could not be retrieved",
-            ) from e
```

Comment:
This error message was misleading as there can be other reasons an exception is raised.